### PR TITLE
Animation controller

### PR DESCRIPTION
**Pull request by maurerdietmar**
Factor out animation controller functionality.

### CODE REVIEW

File: src/animation/interpolate.rs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+// limitations under the License.
+
+use druid::{Color, Insets, Point, Rect, Size, Vec2};
+
+/// Interpolate between two values
+///
+/// Interpolate between `self` and `other` where `value` is the
+/// position (between 0 and 1). For example, a simple linear
+/// interpolation is implemented as: `self + (other - self) * value`
+pub trait Interpolate: PartialEq + Clone {
```
Comment (derekdreery) [inline src/animation/interpolate.rs:22]: Could possibly be better to call this `Lerp` to distinguish between this and other non-linear interpolations.

File: src/animation/controller.rs [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+use super::{AnimationStatus, AnimationDirection};
+
+use crate::RequestCtx;
+
+/// Control animations. An Animation controller produces values between 0.0 and 1.0 during
+/// the given duration. You can run this animation forward, backwards,
+/// altered and repeated.
+pub struct AnimationController {
+    duration: f64,
```
Comment (derekdreery) [inline src/animation/controller.rs:23]: Maybe better to use `std::time::Duration` as it's more descriptive

**Review by derekdreery — APPROVED** [review]
Looks good. Comments are just thoughts, not things that need to be changed.

**Comment by derekdreery** [comment]
@maurerdietmar I've added you as a collaborator so you can merge your own PRs once they're approved (like this one is :slightly_smiling_face: )

File: src/animation/controller.rs (hunk 2 above)
Comment (maurerdietmar) [reply src/animation/controller.rs:23]: Duration is defined as:

```
pub struct Duration {
    secs: u64,
    nanos: u32, // Always 0 <= nanos < NANOS_PER_SEC
}
```
If you use that, you constantly convert from integer to float without any real gain.

**Pull request merged**